raiden-ts/tests/unit/raiden.spec.ts

```diff
@@ -120,9 +124,109 @@ function makeDummyDependencies(): RaidenEpicDeps {
 }
 
 describe('Raiden', () => {
+  const partner = makeAddress();
+
+  function initEpicMock(action$: Observable<RaidenAction>): Observable<raidenSynced> {
+    return action$.pipe(
+      filter(raidenStarted.is),
+      mapTo(
+        raidenSynced({
+          tookMs: 9,
+          initialBlock: 1,
+          currentBlock: 10,
+        }),
+      ),
+    );
+  }
+
   test('address', () => {
     const deps = makeDummyDependencies();
     const raiden = new Raiden(dummyState, deps, combineRaidenEpics(of(dummyEpic)), dummyReducer);
     expect(raiden.address).toBe(dummyState.address);
   });
+
+  function matrixPresenceEpicMock(action$: Observable<RaidenAction>) {
+    return action$.pipe(
+      filter(matrixPresence.request.is),
+      mapTo(
+        matrixPresence.success(
+          {
+            userId: 'John Doe',
+            available: true,
+            ts: 12345,
+          },
+          {
+            address: partner,
+          },
+        ),
+      ),
+    );
+  }
+
+  test('getAvailability', async (done) => {
+    const deps = makeDummyDependencies();
+    const raiden = new Raiden(
+      dummyState,
+      deps,
+      combineRaidenEpics(of(initEpicMock, matrixPresenceEpicMock)),
+      dummyReducer,
+    );
+    await expect(raiden.start()).resolves.toBeUndefined();
+    expect(raiden.started).toEqual(true);
+
+    raiden.action$.pipe(filter(matrixPresence.success.is)).subscribe((action) => {
```

Comment:
Is it possible for you to add a promise pattern here like I did in `stop` ?
https://github.com/raiden-network/light-client/blob/e6c2c46339e9bd4a45cd82c490a6e83939112c7a/raiden-ts/tests/unit/raiden.spec.ts#L162

Reply:
What's the advantage in your eyes?

Reply:
I got a similar review from Andre and wondered the same. Apart from the thing that it looks clean the `.toPromise` function is called only when the observable completes which is the last emitted value each time from the observables

Reply:
I kind of know why Andre is making use of this but i also kind of not like to use this since this is going to be deprecated pretty soon.
https://indepth.dev/posts/1287/rxjs-heads-up-topromise-is-being-deprecated

Reply:
There'll be a drop-in replacement, `lastValueFrom`, so the pattern is still valid. The main benefit of it is that the `expect` calls are performed in the same context (the test function body), instead of a callback (as with `subscribe`), so it's harder to reach the end of the `async test` before all assertions ran. Also, supposedly, it should give better traceback in case of failures, although this isn't always the case with the current node+jest default parameters.

Reply:
Thanks @andrevmatos  for explaining it in a better way !!!

Reply:
✔️